lib/tdf3/src/client/index.ts

```diff
@@ -384,14 +384,23 @@ export class Client {
       eo.attributes.forEach((attr) => s.addJwtAttribute(attr));
       attributeSet = s;
     }
-    encryptionInformation.keyAccess.push(
-      await buildKeyAccess({
-        attributeSet,
-        type: offline ? 'wrapped' : 'remote',
-        url: kasPublicKey.url,
-        kid: kasPublicKey.kid,
-        publicKey: kasPublicKey.publicKey,
-        metadata,
+
+    const splits: SplitStep[] = splitPlan || [{ kas: this.kasEndpoint }];
+    encryptionInformation.keyAccess = await Promise.all(
+      splits.map(async ({ kas, sid }) => {
+        if (!(kas in this.kasKeys)) {
+          this.kasKeys[kas] = fetchKasPublicKey(kas);
```

Comment:
does the cache have a time limit similar to the go one?

Reply:
no it probably should?

Reply:
however this is an existing feature; I'm just changing it to be lazily loaded while before it was loaded during the client constructor

Reply:
https://github.com/opentdf/client-web/issues/326